### PR TITLE
implemented password flag from lxc remote add

### PR DESCRIPTION
**Pull request by benbooth493**
Signed-off-by: Ben Booth <benbooth493@gmail.com>

### CODE REVIEW

File: lxc/remote.go [hunk 1]
```diff
@@ -24,18 +26,20 @@ func (c *remoteCmd) usage() string {
...
+			"lxc remote get-default             	   	    Print the default remote.\n")
 }
 
-func (c *remoteCmd) flags() {}
+func (c *remoteCmd) flags() {
+	gnuflag.StringVar(&c.password, "password", "", gettext.Gettext("Remote admin password"))
+}
 
-func addServer(config *lxd.Config, server string, addr string) error {
+func addServer(config *lxd.Config, server string, addr string, opts map[string]interface{}) error {
```
Comment (tych0) [inline lxc/remote.go]: Is there any reason to use a map here instead of just a `password string`?

Comment (benbooth493) [inline lxc/remote.go]: I was considering the extensibility of any future flags.

Comment (tych0) [inline lxc/remote.go]: Could those not just be additional arguments to the function? The casting seems not so nice.

Comment (stgraber) [inline lxc/remote.go]: Agreed, it's not an external API function so we don't have to care about extending it without changing the function signature, simply adding extra arguments to it is simpler and easier to read/understand.

Comment (benbooth493) [inline lxc/remote.go]: np...fixed

Comment (benbooth493) [inline lxc/remote.go]: Having problems with the pot file. Can you offer any assistance?

Comment (tych0) [inline lxc/remote.go]: On Tue, May 12, 2015 at 07:24:09AM -0700, Ben Booth wrote:
> >  
> > -func addServer(config *lxd.Config, server string, addr string) error {
> > +func addServer(config *lxd.Config, server string, addr string, opts map[string]interface{}) error {
> 
> Having problems with the pot file. Can you offer any assistance?

Yep, you should be able to run 'make i18n' and commit the result (no
merging or other thinking required).

**Comment by stgraber** [comment]
Can you add a test for this feature? Easiest way of testing it would be to alter one of the "lxc remote add" in tests/remote.sh to use --password rather than the echo and sleep trick.

**Comment by benbooth493** [comment]
Sure!

**Comment by stgraber** [comment]
Looks good. Just waiting for Travis to confirm it's all good and I'll merge. Thanks!

**Comment by stgraber** [comment]
Looks like you need another "make i18n"...

**Comment by tych0** [comment]
The "merge master" commit looks a little funny (it duplicates a lot of diff that's on master, I'm not sure how it applies correctly), can you rebase to get rid of that as well?

**Comment by benbooth493** [comment]
Should be good to go. :)

**Comment by stgraber** [comment]
Much better but needs one more rebase according to github (says it can't merge it).

**Comment by benbooth493** [comment]
Should be good to go now.

**Comment by stgraber** [comment]
Sorry, one last thing, would you mind renaming your commit? "merged with master" isn't exactly descriptive of what's being done so it'll be a pain when we release and have to do the release announcement :)

**Comment by tych0** [comment]
I'm a little confused about the commit in general, it looks like it does a bunch of stuff unrelated to actually merging with master. What's the purpose?

**Comment by benbooth493** [comment]
Apologies for the mess....should all be good now. :)

**Comment by stgraber** [comment]
Thanks!

**Pull request merged**